clang/test/SemaSYCL/attr-add-ir-attributes-function-conflict.cpp

```diff
@@ -50,4 +112,67 @@ int main() {
   NonemptyWrapper.kernel_single_task<class NEK9>([]() [[sycl::device_has(sycl::aspect::cpu)]] {});
   // expected-warning@+1 {{kernel has both attribute 'device_has' and kernel properties; conflicting properties are ignored}}
   NonemptyWrapper.kernel_single_task<class NEK10>([]() [[sycl::device_has(sycl::aspect::cpu, sycl::aspect::gpu)]] {});
+
+  // expected-warning@+1 {{kernel has both attribute 'reqd_work_group_size' and kernel properties; conflicting properties are ignored}}
+  NonemptyWrapperWithImplicit.kernel_single_task<class NEKWI1>([]() [[sycl::reqd_work_group_size(1)]] {});
```

Comment:
Did I understand correctly that now when User applied an attribute to `single_task`, a warning will be emitted?

Reply:
Also, the new code in SemaDeclAttr.cpp seem to be skipping the warning, yet it is still emitted here...

Reply:
`Nonempty*` has an explicit property, which means it will have a name-value pair for both "sycl-single-task" and the other mock-property used here, resulting in a total of 4 arguments (excluding the filter) in the `add_ir_attributes_function` attribute. As such it should still warn about potential conflicts, since we know that one of the name-value pairs were caused by an explicit "property".

Reply:
Ok, this is clear now.